doc/modules/neighbors.rst

```diff
@@ -136,9 +136,19 @@ have the same interface; we'll show an example of using the KD Tree here:
 Refer to the :class:`KDTree` and :class:`BallTree` class documentation
 for more information on the options available for nearest neighbors searches,
 including specification of query strategies, distance metrics, etc. For a list
-of available metrics, see the documentation of the :class:`DistanceMetric` class
-and the metrics listed in `sklearn.metrics.pairwise.PAIRWISE_DISTANCE_FUNCTIONS`.
-Note that the "cosine" metric uses :func:`~sklearn.metrics.pairwise.cosine_distances`.
+of valid metrics use :meth:`KDTree.valid_metric` and :meth:`BallTree.valid_metric` as shown here:
+
+    >>> from sklearn.neighbors import KDTree, BallTree
+    >>> dummy_X = np.zeros((1, 1))
+    >>> kdt = KDTree(dummy_X)
+    >>> bt = BallTree(dummy_X)
+    >>> print(f'Valid list of KDTree metrics:\n {kdt.valid_metric}')
+    ['euclidean', 'l2', 'minkowski', 'p', 'manhattan', 'cityblock', 'l1', 'chebyshev', 'infinity']
+    >>> print(f'Valid list of BallTree metrics:\n {bt.valid_metric}')
+    ['euclidean', 'l2', 'minkowski', 'p', 'manhattan', 'cityblock', 'l1', 'chebyshev', 'infinity',
+    'seuclidean', 'mahalanobis', 'wminkowski', 'hamming', 'canberra', 'braycurtis', 'matching',
+    'jaccard', 'dice', 'kulsinski', 'rogerstanimoto', 'russellrao', 'sokalmichener', 'sokalsneath',
+    'haversine', 'pyfunc']
```

Comment:
If using classmethods, this can be changed to:

```suggestion
    >>> from sklearn.neighbors import KDTree, BallTree
    >>> KDTree.valid_metrics())
    ['euclidean', 'l2', 'minkowski', 'p', 'manhattan', 'cityblock', 'l1', 'chebyshev', 'infinity']
    >>> BallTree.valid_metrics())
    ['euclidean', 'l2', 'minkowski', 'p', 'manhattan', 'cityblock', 'l1', 'chebyshev', 'infinity',
    'seuclidean', 'mahalanobis', 'wminkowski', 'hamming', 'canberra', 'braycurtis', 'matching',
    'jaccard', 'dice', 'kulsinski', 'rogerstanimoto', 'russellrao', 'sokalmichener', 'sokalsneath',
    'haversine', 'pyfunc']
```